FreeAPS/Sources/Shortcuts/Bolus/BolusIntentRequest.swift

```diff
@@ -0,0 +1,60 @@
+import Combine
+import CoreData
+import Foundation
+
+@available(iOS 16.0,*) final class BolusIntentRequest: BaseIntentsRequest {
+    private var suggestion: Suggestion? {
+        fileStorage.retrieve(OpenAPS.Enact.suggested, as: Suggestion.self)
+    }
+
+    func bolus(_ bolusAmount: Double) throws -> LocalizedStringResource {
+        var bolusQuantity: Decimal = 0
+        switch settingsManager.settings.bolusShortcut {
+        case .notAllowed:
+            return LocalizedStringResource(
+                "the bolus is not allowed with shortcuts",
+                table: "ShortcutsDetail"
+            )
+        case .limitBolusMax:
+            bolusQuantity = apsManager
```

Comment:
@MikePlante1 is this where the change is needed for your recommendation? Is the check here to compare the bolusAmount with maxBolus?

I think it's something like this:

```
if bolusAmount > maxBolus {
   return LocalizedStringResource("Maximum bolus exceeded, no bolus administered.")
}
else {
 bolusQuantity = roundBolus(Decimal(bolusAmount))
}
```

Do this down below for the other version - checking for recommended percentage.